docs/book/src/reference/owner_references.md

```diff
@@ -0,0 +1,79 @@
+2. Owner references should be re-reconciled if they are lost for an object. This is required as some tools - e.g. velero - may delete owner references on objects.
+3. Owner references should be kept to the most recent apiVersion.
+   - This ensures garbage collection still works after an old apiVersion is no longer served.
+4. Owner references should not be added unless required.
+   - Multiple owner references on a single object should be exceptional.
+
+
+  
+
+## Owner reference relationships in Cluster API
+
+The below tables map out the a reference for ownership relationships for the objects in a Cluster API cluster.
+Note: Providers may implement their own ownership relationships which may or may not map directly to the below tables.
+
+## Kubernetes core types
+
+| type      | Owner               | Note                     |
+|-----------|---------------------|--------------------------|
+| Secret    | KubeadmControlPlane | For cluster certificates |
+| Secret    | KubeadmConfig       | For bootstrap secrets    |
+| Secret    | ClusterResourceSet  | When created by CRS      |
+| ConfigMap | ClusterResourceSet  |                          |
+
+## Core types
+
+| type                | Owner             | Note                       |
+|---------------------|-------------------|----------------------------|
+| ExtensionConfig     | None              |                            |
+| ClusterClass        | None              |                            |
+| Cluster             | None              |                            |
+| MachineDeployments  | Cluster           |                            |
+| MachineSet          | MachineDeployment |                            |
+| Machine             | MachineSet        | When created by MachineSet |
+| Machine             | ControlPlane      | When created by KCP        |
+| MachineHealthChecks | Cluster           |                            |
+
+
+
+## Experimental types
+| type                       | Owner              | Note |
+|----------------------------|--------------------|------|
+| ClusterResourcesSet        | None               |      |
+| ClusterResourcesSetBinding | ClusterResourceSet |      |
+| MachinePool                | Cluster            |      |
+
+
+## KubeadmControlPlane types
+| type                        | Owner        | Note |
+|-----------------------------|--------------|------|
+| KubeadmControlPlane         | Cluster      |      |
+| KubeadmControlPlaneTemplate | ClusterClass |      |
+    
+
+## Kubeadm bootstrap types
+| type                  | Owner        | Note                            |
+|-----------------------|--------------|---------------------------------|
+| KubeadmConfig         | Machine      | When created by MachineSet      |
+| KubeadmConfig         | MachinePool  | When created by MachinePool     |
```

Comment:
MachinePool doesn't create the KubeadmConfig. It must be created before and is then just referenced (sort of like a KubeadmConfigTemplate for a MachineDeployment)